syntax = "proto3";

package replication;

service ReplicationService {
  rpc Read(ReadRequest) returns (ReadResponse);
  rpc Write(WriteRequest) returns (WriteResponse);
  rpc PropagateOtherLayers(PropagateOtherLayersRequest) returns (PropagateOtherLayersResponse);
  rpc PropagateCoreNodes(PropagateCoreNodesRequest) returns (PropagateCoreNodesResponse);
}

message ReadRequest {
  int32 key = 1;
  string id = 2;
}

message ReadResponse {
  int32 valor = 1;
  bool ack = 2;
  string missatge = 3;
}

message WriteRequest {
  int32 key = 1;
  int32 valor = 2;
  string id = 3;
}

message WriteResponse {
  bool ack = 1;
  string missatge = 2;
}

message PropagateOtherLayersRequest {
  map<int32, int32> data = 1;
  int32 versio = 2;
}

message PropagateOtherLayersResponse {
  bool ack = 1;
}

message PropagateCoreNodesRequest {
  map<int32, int32> data = 1;
  int32 versio = 2;
  string id_send = 3;
}

message PropagateCoreNodesResponse {
  bool ack = 1;
}